### PR TITLE
Add ID localization of force delete stateful pod

### PR DESCRIPTION
**Pull request by danninov**
Add ID localization of task force delete statefulset pod. Issue #22296

### CODE REVIEW

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+* Ini merupakan tugas yang cukup rumit dan memiliki potensi melanggar beberapa properti yang melekat dari StatefulSet.
+* Sebelum melanjutkan, pastikan kamu paham dengan pertimbangan yang disebutkan di bawah ini.
+
+
+
+<!-- steps -->
+
+## Pertimbangan StatefulSet
+
+Pada operasi normal dari StatefulSet, **tidak pernah** ada kebutuhan untuk menghapus paksa sebuah Pod StatefulSet. [Pengontrol StatefulSet](/docs/concepts/workloads/controllers/statefulset/) bertanggung jawab terhadap pembuatan, _scaling_ dan penghapusan terhadap anggota dari StatefulSet. Pengontrol akan berusaha menjaga agar jumlah Pod yang ditentukan dari 0 hingga N-1 hidup dan siap sedia. StatefulSet memastikan bahwa, pada waktu kapanpun, akan ada minimal satu Pod dengan identitas yang telah ditetapkan berjalan pada klaster. Hal ini direferensikan sebagai semantik *at most one* yang disediakan StatefulSet.
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:24]: ID Docs is available.
Scaling can be translated into this https://kbbi.kemdikbud.go.id/entri/Penyekalaan
Nit! I prefer to keep “controller”
```suggestion
Pada operasi normal dari StatefulSet, **tidak pernah** ada kebutuhan untuk menghapus paksa sebuah Pod StatefulSet. [_Controller_ StatefulSet](/id/docs/concepts/workloads/controllers/statefulset/) bertanggung jawab terhadap pembuatan, penyekalaan, dan penghapusan terhadap anggota dari StatefulSet. _Controller_ akan berusaha menjaga agar jumlah Pod yang ditentukan dari 0 hingga N-1 hidup dan siap sedia. StatefulSet memastikan bahwa, pada waktu kapanpun, akan ada minimal satu Pod dengan identitas yang telah ditetapkan berjalan pada klaster. Hal ini direferensikan sebagai semantik *at most one* yang disediakan dalam StatefulSet.
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+Penghapusan paksa secara manual harus dilakukan dengan hati-hati, karena hal tersebut berpotensi melanggar semantik _at most one_ yang melekat pada StatefulSet. StatefulSet dapat digunakan untuk menjalankan aplikasi terklaster dan terdistribusi yang membutuhkan identitas jaringan dan penyimpanan yang stabil dan tetap. Aplikasi-aplikasi ini biasanya memiliki konfigurasi yang tergantung dengan sejumlah anggota dengan identitas yang tetap. Memiliki banyak anggota dengan identitas yang sama berpotensi menjadi bencana dan dapat menimbulkan kehilangan data (contohnya skenario  _split brain_ pada sistem berbasis kuorum).
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:26]: “Bencana” seems to much.
```suggestion
Penghapusan paksa secara manual harus dilakukan dengan hati-hati, karena hal tersebut berpotensi melanggar semantik _at most one_ yang melekat pada StatefulSet. StatefulSet dapat digunakan untuk menjalankan aplikasi terklaster dan terdistribusi yang membutuhkan identitas jaringan dan penyimpanan yang stabil dan tetap. Aplikasi-aplikasi ini biasanya memiliki konfigurasi yang tergantung dengan sejumlah anggota dengan identitas yang tetap. Memiliki banyak anggota dengan identitas yang sama berpotensi menjadi masalah dan dapat menimbulkan kehilangan data (contohnya skenario  _split brain_ pada sistem berbasis kuorum).
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Menghapus Pod
+
+Kamu dapat melakukan penghapusan pod dengan _graceful_ dengan perintah berikut:
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:30]:
```suggestion
Kamu dapat melakukan penghapusan Pod secara _graceful_ dengan perintah berikut:
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+```shell
+kubectl delete pods <pod>
+```
+
+Agar perintah di atas mengarah ke terminasi dengan _graceful_, Pod **tidak boleh** menspesifikasikan `pod.Spec.TerminationGracePeriodSeconds` dengan nilai 0. Praktik untuk mengatur nilai `pod.Spec.TerminationGracePeriodSeconds` menjadi 0 detik merupakan hal yang tidak aman dan sangat tidak disarankan untuk Pod StatefulSet. Penghapusan dengan _graceful_  itu aman dan akan memastikan bahwa Pod [akan mati dengan _gracefully_](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination) sebelum kubelet menghapus nama dari apiserver.
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:36]:
```suggestion
Agar perintah di atas mengarah ke terminasi secara _graceful_, Pod **tidak boleh** menspesifikasikan `pod.Spec.TerminationGracePeriodSeconds` dengan nilai 0. Praktik untuk mengatur nilai `pod.Spec.TerminationGracePeriodSeconds` menjadi 0 detik merupakan hal yang tidak aman dan sangat tidak disarankan untuk Pod StatefulSet. Penghapusan secara _graceful_  itu aman dan akan memastikan bahwa Pod [akan mati dengan _gracefully_](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination) sebelum kubelet menghapus nama dari apiserver.
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Kubernetes (versi 1.5 atau lebih baru) tidak akan menghapus Pod hanya karena Node tidak dapat dijangkau. Pod yang berjalan pada Node yang tidak dapat dijangkau akan memasuki keadaan 'Terminating' atau 'Unknown' setelah [waktu habis](/docs/admin/node/#node-condition). Pod juga dapat memasuki keadaan ini saat pengguna berusaha melakukan penghapusan dengan _graceful_ terhadap Pod pada Node yang tidak dapat dijangkau. Cara yang hanya dapat dilakukan untuk menghapus Pod pada keadaan tersebut dari apiserver adalah:
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:38]: Link to ID Doc.
```suggestion
Kubernetes (versi 1.5 atau lebih baru) tidak akan menghapus Pod hanya karena Node tidak dapat dijangkau. Pod yang berjalan pada Node yang tidak dapat dijangkau akan memasuki keadaan 'Terminating' atau 'Unknown' setelah [waktu habis](/id/docs/admin/node/#node-condition). Pod juga dapat memasuki keadaan ini saat pengguna berusaha melakukan penghapusan dengan _graceful_ terhadap Pod pada Node yang tidak dapat dijangkau. Cara yang hanya dapat dilakukan untuk menghapus Pod pada keadaan tersebut dari apiserver adalah:
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+   * Objek Node telah dihapus (baik oleh kamu, atau oleh [Pengontrol Node](/docs/admin/node)).<br/>
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:40]: Link to ID Doc.
```suggestion
   * Objek Node telah dihapus (baik oleh kamu, atau oleh [Pengontrol Node](/id/docs/admin/node)).<br/>
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+   * Kubelet pada Node yang tidak responsif akan menanggapi, lalu mematikan Pod dan menghapusnya dari apiserver.<br/>
+   * Penghapusan paksa Pod oleh pengguna.
+
+Praktik terbaik yang direkomendasikan adalah menggunakan pendekatan pertama atau kedua. Jika sebuah Node telah terkonfirmasi mati (contohnya terputus dari jaringan secara permanen, dimatikan, dll), lalu menghapus objek Node. Jika Node mengalami partisi jaringan, maka coba selesaikan masalah ini atau menunggu masalah itu terselesaikan. Saat partisi terselesaikan, kubelet akan menyelesaikan penghapusan Pod serta membebaskan namanya dari apiserver.
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:44]:
```suggestion
Praktik terbaik yang direkomendasikan adalah menggunakan pendekatan pertama atau kedua. Jika sebuah Node telah terkonfirmasi mati (contohnya terputus dari jaringan secara permanen, dimatikan, dll), maka objek Node dihapus. Jika Node mengalami partisi jaringan, maka coba selesaikan masalah ini atau menunggu masalah itu terselesaikan. Saat partisi terselesaikan, kubelet akan menyelesaikan penghapusan Pod serta membebaskan namanya dari apiserver.
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+Normalnya, sistem akan menyelesaikan penghapusan saat Pod tidak lagi berjalan pada Node, atau Node telah dihapus oleh administrator. Kamu dapat mengabaikan hal ini dengan menghapus paksa Pod.
+
+### Penghapusan Paksa
+
+Penghapusan paksa **tidak** menunggu konfirmasi dari kubelet bahwa Pod telah diterminasi. Terlepas dari apakah penghapusan paksa sukses mematikan sebuah Pod, namanya akan segera dibebaskan dari apiserver. Hal ini berakibat pengontrol StatefulSet akan membuat Pod pengganti dengan identitas yang sama; ini dapat menimbulkan duplikasi terhadap Pod apabila ternyata Pod tersebut masih berjalan, dan jika Pod tersebut masih dapat berkomunikasi dengan anggota Statefulset lainnya, hal ini berakibat terjadi pelanggaran semantik _at most one_ dari StatefulSet yang telah dijamin.
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:50]:
```suggestion
Penghapusan paksa **tidak** menunggu konfirmasi dari kubelet bahwa Pod telah diterminasi. Terlepas dari apakah penghapusan paksa sukses mematikan sebuah Pod, namanya akan segera dibebaskan dari apiserver. Hal ini berakibat _controller_ StatefulSet akan membuat Pod pengganti dengan identitas yang sama; ini dapat menimbulkan duplikasi terhadap Pod apabila ternyata Pod tersebut masih berjalan, dan jika Pod tersebut masih dapat berkomunikasi dengan anggota Statefulset lainnya, hal ini berakibat terjadi pelanggaran semantik _at most one_ dari StatefulSet yang telah dijamin.
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+kubectl delete pods <pod> --grace-period=0 --force
+```
+
+Jika kamu menggunakan kubectl <= 1.4, kamu harus menghilangkan pilihan `--force` dan gunakan:
+
+```shell
+kubectl delete pods <pod> --grace-period=0
+```
+
+Jika setelah perintah ini dijalankan dan pod tetap berada pada kondisi `Unknown`, gunakan perintah berikut untuk menghapus pod dari klaster:
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:66]:
```suggestion
Jika setelah perintah ini dijalankan dan Pod tetap berada pada kondisi `Unknown`, gunakan perintah berikut untuk menghapus Pod dari klaster:
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+Selalu jalankan penghapusan paksa Pod StatefulSet dengan hati-hati dan penuh pemahaman terhadap risiko yang dapat timbul.
+
+
+
+## {{% heading "whatsnext" %}}
+
+
+Pelajari lebih lanjut [debugging StatefulSet](/docs/tasks/debug-application-cluster/debug-stateful-set/).
```
Comment (ariscahyadi) [inline content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:79]:
```suggestion
Pelajari lebih lanjut [_debugging_ StatefulSet](/id/docs/tasks/debug-application-cluster/debug-stateful-set/).
```

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md (hunk 2 above)
Comment (danninov) [reply content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:26]: I'll change to `berpotensi menimbulkan kerusakan dan kehilangan data`

File: content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md (hunk 10 above)
Comment (danninov) [reply content/id/docs/tasks/run-application/force-delete-stateful-set-pod.md:79]: There is no ID localization yet for task debugging.

**Comment by irvifa** [comment]
Awesome, thanks for the contribution 😎 
/approve

**Pull request merged**